bindings/pydeck/pydeck/bindings/deck.py

```diff
@@ -102,9 +104,7 @@ def __init__(
         self.map_provider = str(map_provider).lower() if map_provider else None
         self.deck_widget.map_provider = map_provider
         if self.mapbox_key is None and self.map_provider == Providers.MAPBOX:
```

Comment:
This check should be done on the JS side in the playground. 

If you want to present the error in python, use the back channel and send a `deck-error` or `deck-warning` event to python.